modules/hivemq-edge-module-modbus/src/main/java/com/hivemq/edge/adapters/modbus/config/PollingContextImpl.java

```diff
@@ -74,14 +79,52 @@ public PollingContextImpl(
             @JsonProperty("includeTimestamp") final @Nullable Boolean includeTimestamp,
             @JsonProperty("includeTagNames") final @Nullable Boolean includeTagNames,
             @JsonProperty("userProperties") final @Nullable List<UserProperty> userProperties,
-            @JsonProperty(value = "addressRange", required = true) final @NotNull AddressRange addressRange) {
+            @JsonProperty(value = "addressRange", required = true) final @NotNull AddressRange addressRange,
+            @JsonProperty(value = "dataType") final @Nullable ModbusDataType dataType) throws ProtocolAdapterException {
         this.destination = destination;
         this.qos = requireNonNullElse(qos, 0);
         this.messageHandlingOptions = requireNonNullElse(messageHandlingOptions, MQTTMessagePerSubscription);
         this.includeTimestamp = requireNonNullElse(includeTimestamp, true);
         this.includeTagNames = requireNonNullElse(includeTagNames, false);
         this.addressRange = addressRange;
         this.userProperties = requireNonNullElseGet(userProperties, List::of);
+        this.dataType = requireNonNullElse(dataType, ModbusDataType.INT_16);
+
+        final int registerCount = addressRange.endIdx - addressRange.startIdx;
+        switch (this.dataType) {
+            case INT_16:
+            case UINT_16:
+                if (registerCount != 1) {
```

Comment:
do we need a change in the frontend so this is cannot happen ?

Reply:
Hmm, I think yes.